src/gui/accountstate.cpp

```diff
@@ -136,8 +137,18 @@ AccountState::AccountState(AccountPtr account)
                 break;
             }
         });
-    } else {
-        qCWarning(lcAccountState) << "Failed to load QNetworkInformation";
+
+        connect(qNetInfo, &QNetworkInformation::isMeteredChanged, this, [this](bool isMetered) {
+            if (ConfigFile().pauseSyncWhenMetered()) {
+                if (state() == State::Connected && isMetered) {
+                    qCInfo(lcAccountState) << "Network switched to a metered connection, setting account state to PausedDueToMetered";
```

Comment:
I guess we need a new icon, currently it looks like we are logged out.
Maybe we should display the paused icon for now?
![image](https://github.com/owncloud/client/assets/200626/92ed91d9-e7a2-40fa-a806-bc0f678ee7ae)
![image](https://github.com/owncloud/client/assets/200626/d724635b-3ef6-428d-9a82-ac67459ed590)